cloud/etc/deploy-maas-region/main.tf

```diff
@@ -45,7 +45,39 @@ resource "juju_application" "maas-region" {
   config = var.charm_maas_region_config
 }
 
-resource "juju_integration" "maas-region-postgresql" {
+resource "juju_application" "pgbouncer" {
+  name  = "pgbouncer"
+  model = data.juju_model.machine_model.name
+  units = 0 # it is a subordinate charm
+
+  charm {
+    name     = "pgbouncer"
+    channel  = var.charm_pgbouncer_channel
+    revision = var.charm_pgbouncer_revision
+    base     = "ubuntu@22.04"
+  }
+
+  config = merge({
+    pool_mode          = "transaction"
+    max_db_connections = floor(90 / max(length(var.machine_ids), 1))
```

Comment:
Will this dynamically resize?

Reply:
Whenever a new anvil node is joining the cluster with role `region` the list of `machine_ids` will be updated. As such, the `max_db_connections` will be updated in Terraform. And, in the end, that means the juju will re-configure the pgbouncer charmed application. Based on its implementation, the pgbouncer charm code will handle the config change and will apply it in the proper way.